paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -133,4 +141,43 @@ public boolean branchExists(String branchName) {
         Path branchPath = branchPath(branchName);
         return fileExists(branchPath);
     }
+
+    /** Get branch count for the table. */
+    public long branchCount() {
+        try {
+            return listVersionedFileStatus(fileIO, branchDirectory(), BRANCH_PREFIX).count();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** Get all branches for the table. */
+    public List<TableBranch> branches() {
```

Comment:
I see this in this pr https://github.com/apache/incubator-paimon/pull/2754 and return different types, they may conflict before merging.

Reply:
Fixed